README.md
MeatyBones V1.1
=================

MeatyBones is a WordPress base framework that is similar to the <a href="http://themble.com/bones/" target="_blank">Bones</a> framework in that it is a clean slate ready for an asthetic makeover. MB is an improvement in that there are more configurable options and features included and many common widget and content areas are included as well. The latest iteration of MeatyBones is using the 1% Grid developed by <a href="http://onepcssgrid.mattimling.com/" target="_blank">Matt Imling</a>. This grid has been included in page/post templates ready for use as well as the home page template.


Demos
=====

<ul>
<li>
<a href="http://meatybones-1000.alwaysup2late.com/" target="_blank">1000px Base Version</a>
</li>
<li>
<a href="http://meatybones-1200.alwaysup2late.com/" target="_blank">1200px Base Version</a>
</li>
</ul>

Features
========

<ul>
  <li>HTML5/CSS3</li>
  <li>IE7/8 Friendly*</li>
  <li>Responsive/Fluid</li>
  <li>Customizable homepage slider</li>
  <li>Customizable content zones</li>
  <li>Adaptable layout to meet content requirements</li>
  <li>Automatic creation of mobile select nav</li>
  <li>jQuery-Less, using pure JavaScript</li>
  <li>SASS Logic and config.rb file included</li>
</ul>

Installation
============

<strong>Within WordPress</strong>
<ol>
<li>Download the master zip file</li>
<li>Rename it to whatever you want (leave the .zip at the end)</li>
<li>In WordPress Admin, goto "Appearance" -> "Themes"</li>
<li>Click "Install Themes"</li>
<li>Click "Upload"</li>
<li>Locate your renamed zip file and click "Install Now"</li>
<li>Under "Manage Themes", click "Activate" for your new theme</li>
</ol>

<strong>Within Themes Folder</strong>
<ol>
<li>Download the master zip file</li>
<li>Unzip and rename to whatever you want</li>
<li>Move the newly renamed folder to [basedir]/wp-content/themes/</li>
<li>Under "Manage Themes", click "Activate" for your new theme</li>
</ol>

Layout Options
==============

<ul>
<li>Homepage (Configure from "Theme Options" panel)</li>
<li>Content Full (100%)</li>
<li>Content/Sidebar (50%/50%)</li>
<li>Content/Sidebar (60%/40%)</li>
<li>Content/Sidebar (70%/30%)</li>
<li>Content/Sidebar (80%/20%)</li>
<li>Sidebar/Content (80%/20%)</li>
<li>Sidebar/Content (70%/30%)</li>
<li>Sidebar/Content (60%/40%)</li>
<li>Sidebar/Content (50%/50%)</li>
</ul>

Widget Areas
============

<ul>
<li>Header</li>
<li>Sidebar</li>
<li>Footer Left</li>
<li>Footer Center</li>
<li>Footer Right</li>
</ul>

Options Panel
=============

<strong>Base Width Settings</strong>
<ul>
<li>1000px</li>
<li>1200px</li>
</ul>

<strong>Logo Settings</strong>
<ul>
<li>Logo Image</li>
<li>Logo Alt Text</li>
</ul>

<strong>Social Links (Widget)</strong>
<ul>
<li>Main Email Address</li>
<li>Default Subject for Email</li>
<li>Twitter Username (ex: @twitter)</li>
<li>Facebook link</li>
<li>YouTube Link</li>
<li>LinkedIn link</li>
<li>Display RSS Link</li>
</ul>

<strong>Slider Settings</strong>
<ul>
<li>Display Slider on Homepage</li>
<li>Display Slider Previous/Next Buttons</li>
<li>Display Slider Index Controls</li>
</ul>

<strong>Homepage Boxes</strong>
<ul>
<li>Home Box1 Page</li>
<li>Home Box1 "read more" link</li>
<li>Homepage Box2 Page</li>
<li>Home Box2 "read more" link</li>
<li>Home Box3 Page</li>
<li>Home Box3 "read more" link</li>
<li>Home Box4 Page</li>
<li>Home Box4 "read more" link</li>
<li>Display Homebox Titles</li>
</ul>

<strong>Homepage Blurb (Below Slider)</strong>
<ul>
<li>Display Homepage Blurb</li>
<li>Display Blurb Title</li>
<li>Blurb Content page</li>
</ul>	

<strong>Copyright Info (Footer)</strong>
<ul>
<li>Copyright Text (Can be HTML)</li>
</ul>

<strong>Misc</strong>
<ul>
<li>Meta Keywords (Head)</li>
<li>Meta Description (Head)</li>
<li>Tracking script (Footer)</li>
</ul>

CHANGELOG.md
<strong>V1.0.0</strong>
<ul>
<li>Initial build of base theme and options panel</li>
</ul>

<strong>V1.0.1</strong>
<ul>
<li>Added ability to change base width from 1000px to 1200px from the Theme Options panel</li>
</ul>

<strong>V1.0.2</strong>
<ul>
<li>Changed swipe start slide setting so if only one image is uploaded it will display</li>
</ul>

<strong>V1.0.3</strong>
<ul>
<li>Added CSS viewport settings (forward thinking)</li>
<li>Added correct meta viewport to head</li>
<li>Added CSS file for grid examples in the readme folder</li>
<li>Added HTML5Shiv to js/vendor</li>
<li>Added Page name to body class for all pages</li>
</ul>

<strong>V1.0.4</strong>
<ul>
<li>Fixed functions.php reference to blurb variable</li>
<li>Added comments to SASS includes files</li>
<li>Fixed tacking code integration into footer</li>
</ul>

<strong>V1.0.5</strong>
<ul>
<li>Added missing > to body tag</li>
<li>Added previous/next support to slider</li>
<li>Added index control support to slider</li>
<li>Added slider specific section to Theme Options</li>
<li>Only showing prev/next & index dots on larger screen sizes</li>
<li>Added slider caption support via except field on post edit screen</li>
</ul>

<strong>V1.0.6</strong>
<ul>
<li>Fixed unclosed tag in header.php</li>
<li>Added "innerPage" body class to inner pages</li>
</ul>

<strong>V1.0.7</strong>
<ul>
<li>Added a fix for nav links for sliders</li>
</ul>

<strong>V1.0.8</strong>
<ul>
<li>Updated HTML5 elements</li>
<li>Included a custom Connect Widget</li>
<li>Added more config options to Theme Options</li>
</ul>

<strong>V1.0.9</strong>
<ul>
<li>Modulated functions.php</li>
<li>Removed symbols from read more links</li>
<li>Added support for featured images at the top of pages</li>
</ul>

<strong>V1.1</strong>
<ul>
<li>Added YouTube support to MB Connect Widget</li>
</ul>

footer.php
		<div class="clearfix"></div>
	</section>
	<!-- End .wrapper -->
	<footer>
		<?php if (get_option('mb_base_width_setting') == 'large') {?>
			<section class="onepcssgrid-1200">
		<?php } else {?>
			<section class="onepcssgrid-1000">
		<?php }?>		
			<?php if (function_exists('is_sidebar_active') && is_sidebar_active('footer_left')) { ?>
				<?php if (is_sidebar_active('footer_left') && !is_sidebar_active('footer_center') && !is_sidebar_active('footer_right')) { ?>
					<div class="col12">
				<?php } elseif (is_sidebar_active('footer_left') && is_sidebar_active('footer_center') && !is_sidebar_active('footer_right')) {?>
					<div class="col6">
				<?php } else {?>
					<div class="col4">
				<?php }?>
					<?php dynamic_sidebar('footer_left'); ?>
				</div>
				<?php if (is_sidebar_active('footer_center')) { ?>
					<?php if (is_sidebar_active('footer_left') && is_sidebar_active('footer_center') && !is_sidebar_active('footer_right')) {?>
					     <div class="col6 last">
					<?php } else {?>
					     <div class="col4">
					<?php }?>
						<?php dynamic_sidebar('footer_center'); ?>
					</div>
				<?php } ?>
				<?php if (is_sidebar_active('footer_right')) { ?>
					<?php if (is_sidebar_active('footer_left') && is_sidebar_active('footer_center') && !is_sidebar_active('footer_right')) {?>
					     <div class="col6">
					<?php } else {?>
					     <div class="col4 last">
					<?php }?>
						<?php dynamic_sidebar('footer_right'); ?>
					</div>
				<?php } ?>
			<?php } ?>
			<div class="copyright col12">
				<?php 
					if (get_option('mb_copyright') <> "") {
						echo stripslashes(get_option('mb_copyright'));
					}
				?>
			</div>

			<div class="clearfix"></div>
		</section>
	</footer>
<?php wp_footer(); ?>
<script type="text/javascript" src="<?php bloginfo('template_directory'); ?>/js/vendor/selectnav.js"></script>
<script type="text/javascript" src="<?php bloginfo('template_directory'); ?>/js/global.js"></script>
<?php if (get_option('mb_slider_enable') == 'yes' && is_front_page()) {?>
<script type="text/javascript" src="<?php bloginfo('template_directory'); ?>/js/vendor/swipe.js"></script>
<script type="text/javascript">
	setTimeout(function () {
		window.mySwipe = new Swipe(document.getElementById("slider"), {
			startSlide: 0,
			speed: 400,
			auto: 3000,
			continuous: true,
			disableScroll: false,
			stopPropagation: false,
			callback: function(index, elem) {
				updateActiveDot();
			},
			transitionEnd: function(index, elem) {}
		});
	}, 100);

	<?php if (get_option('mb_slider_prevnext') == "yes") {?>
		var prevbutton = document.getElementById('prev'),
			nextbutton = document.getElementById('next');

		prevbutton.onclick = function(event) {
			event.preventDefault();
			mySwipe.prev();
		}

		nextbutton.onclick = function(event) {
			event.preventDefault();
			mySwipe.next();
		}
	<?php }?>

	<?php if (get_option('mb_slider_dots') == "yes") {?>
		setTimeout(function() {
			var slideControls = document.getElementById('slideControls'),
				numberSlides = mySwipe.getNumSlides();

			for (var i=0; i < numberSlides; i++) {
				var item = document.createElement("li");
				
				item.innerHTML = '<a href="#">'+ [i] + '</a>';

				slideControls.appendChild(item);

				var firstLi = slideControls.childNodes[0];
				firstLi.classList.add("active")
			}

			getEventTarget = function(event) {
				event = event || window.event;
				return event.target || event.srcElement; 
			}

			slideControls.onclick = function(event) {
				var target = getEventTarget(event);

				event.preventDefault();
				
				if (target.tagName == 'A') {
					mySwipe.slide(target.innerHTML, 400);
				}
			};

			updateActiveDot = function() {
				var dots = document.getElementById('slideControls').getElementsByTagName('li'),
					position = mySwipe.getPos();

				var i = dots.length;
				while (i--) {
					dots[i].className = '';
				}

				dots[position].className = 'active';
			}
		}, 100);
	<?php }?>
</script>
<?php }?>
<?php 
	if (get_option('mb_analytics') <> "") { 
		echo stripslashes(stripslashes(get_option('mb_analytics'))); 
	}
?>
</body>
</html>